### ISSUE TITLE
MD029 rule configuration not working

### ISSUE THREAD

**Issue by Owen-OptiGrid** [rvben/rumdl#16]
I have this in my .rundl.toml:

```toml
[MD029]
style = "ordered"
```

but when running rumdl I get this error:

```shell
rumdl check .
[config warning] Unknown option for rule MD029: style
```

I'm using:

```shell
rumdl --version
rumdl 0.0.102
```

**Comment by rvben** [comment]
Hi @Owen-OptiGrid ,

Thanks for raising this issue. That really helps improving rumdl. 
I found out the a couple more rules had similar issues that I also fixed. This issue and the similar issues were fixed in 0.0.103.

Please let me know if you hit any other issues.

**Comment by Owen-OptiGrid** [comment]
Hi @rvben - this issue still isn't resolved. You fixed the config parse error but you haven't fixed the behaviour when I configure `style = "ordered"` - it still acts like `style = "one"` and continues enforcing and auto fixing according to the `one` behaviour.

I'm testing with rumdl 0.0.104

**Comment by rvben** [comment]
Hi @Owen-OptiGrid,

I've re-tested it with a simple example:

```markdown
# Hello World

1. Hello

1. World
```

```toml
[MD029]
style = "ordered"
```

```bash
$ rumdl check .      
README.md:5:1: [MD029] Ordered list item number 1 does not match style (expected 2) [*]

Issues: Found 1 issues in 1 file (20ms)
Run with `--fix` to automatically fix 1 of the 1 issue
```

Also the fix seems to work as expected. 

Maybe there's a deeper issue or something else going on. Can you maybe share the output of `rumdl config` to check where the configuration is coming from and if it's picked up correctly?


Kind regards,

Ruben

**Comment by Owen-OptiGrid** [comment]
Interesting... yeah I'll test again and confirm it picked up the config.

**Comment by Owen-OptiGrid** [comment]
So I dug deeper and the issue is around multi-line items and indentation and whether a collection of lines is recognised as one item or two...

So with ordered style select this doesn't flag any error (perhaps rightly so):

```markdown
# Test Rumdl

1. Blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah
blah
1. Blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah
blah
```

This also doesn't get flagged as an error... but I think it should be:

```markdown
# Test Rumdl

1. Blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah
   blah
1. Blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah
   blah
```

Finally this is picked up as an error:

```markdown
# Test Rumdl

1. Blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah
    blah
1. Blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah
    blah
```

and gets auto-corrected to this:

```markdown
# Test Rumdl

1. Blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah
    blah
2. Blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah blah
    blah
```

The funny thing is even with no indentation of subsequent lines of the same item... both the markdown renders I tried in VS Code (and GitHub itself) still seem to recognise this as a single markdown list (not two separate ones), e.g.

**Markdown Preview Enhanced**

<img width="1706" height="668" alt="Image" src="https://github.com/user-attachments/assets/b9c90f0a-010d-4024-a2b6-f7eac6957a1e" />

**Markdown All in One**

<img width="1619" height="230" alt="Image" src="https://github.com/user-attachments/assets/0fc80ae0-a76f-4e05-a212-9ededdcf4211" />

I don't know what the correct behaviour exactly should be for recognising multi-line single element list items... although I'd suggest you probably want to match however the majority of markdown renderers... especially how GitHub chooses to parse it for rendering.

**Comment by Owen-OptiGrid** [comment]
I have to thank you too @rvben - I got a shout out the PythonBytes podcast for mentioning rumdl. Hope you get a lot more stars, this repo deserves it: https://pythonbytes.fm/episodes/show/443/patching-multiprocessing

**Comment by rvben** [comment]
@Owen-OptiGrid , thanks! That's the first shout-out I've seen, that's amazing. 

I've added some more fixes to the latest release. The first example, however, I need to look at a bit more on how to properly approach that to make it user-friendly.

**Comment by rvben** [comment]
I think it should now be fixed. Can you check and see if it's as you would expect, @Owen-OptiGrid ?

**Comment by Owen-OptiGrid** [comment]
Sorry for the slow response - I had noticed the change. Yeah happy for it to be closed.
